### PR TITLE
feat(dropdown): add dropdown header

### PR DESCRIPTION
**Pull request by alejoviola**
[Associated Notion Card](https://www.notion.so/rational-nomads/Add-header-to-Dropdown-27fe8a91822f4e17a112b5fd112f8536?pvs=4)

-----

## 1. Pull request checklist

- [x] Notion card has been moved to the Code Review column
- [x] Notion card has a link to this PR
- [x] A reviewer has been assigned to the Notion card

## 2. PR type
- Feature

## 3. What is the old behaviour?
- DropdownMenu component doesn't have a header and children support.

## 4. What is the new behaviour?
- Create DropdownHeader component to be put inside DropdownMenu.
- Add children support on DropdownMenu component.

## 5. Other information
https://www.loom.com/share/67bfdaab108c49e68dcda2c037e78a5e

### CODE REVIEW

**Comment by pnagarjun16** [comment]
Instead of creating the Dropdown header component separately, we only have to implement the component inside the Dropdown menu. As mentioned in this https://www.loom.com/share/917dde6b977f4ff4879da69a18d409f8

**Review by pnagarjun16 — CHANGES_REQUESTED** [review]

File: src/components/DropdownMenu/DropdownMenu.tsx [hunk 1]
```diff
@@ -87,3 +91,25 @@ export const DropdownMenu = forwardRef<HTMLDivElement, DropdownMenuProps>(
     );
   }
 );
+
+export interface HeaderProps {
+  /** Item to render inside the Header */
+  item: string | ReactNode;
```
Comment (pnagarjun16) [inline src/components/DropdownMenu/DropdownMenu.tsx:97]: change to `item: ReactNode`

File: src/components/DropdownMenu/DropdownMenu.scss [hunk 2]
```diff
@@ -12,29 +13,46 @@
...
   .zui-dropdown-item {
     padding: 1rem;
 
-    color: #{$text-primary};
+    color: $text-primary;
     outline: none;
 
     &:hover {
-      background: #381C4E;
+      background: #381c4e;
```
Comment (pnagarjun16) [inline src/components/DropdownMenu/DropdownMenu.scss:31]: Can we move this one to variables?

File: src/components/DropdownMenu/DropdownMenu.tsx [hunk 3]
```diff
@@ -87,3 +91,25 @@ export const DropdownMenu = forwardRef<HTMLDivElement, DropdownMenuProps>(
...
+export const Header: FC<HeaderProps> = ({ item, className, titleClassName }) => {
+  const itemElement =
+    typeof item === 'string' ? (
+      <span className={classNames('zui-dropdown-header-title', titleClassName)}>{item}</span>
+    ) : (
+      item
+    );
+
+  return <div className={classNames('zui-dropdown-header-container', className)}>{itemElement}</div>;
+};
```
Comment (pnagarjun16) [inline src/components/DropdownMenu/DropdownMenu.tsx:115]: Can we update it like this 
```return <div className={classNames('zui-dropdown-header-container', className)} >{item}</div>```

File: src/components/DropdownMenu/DropdownMenu.tsx [hunk 4]
```diff
@@ -24,6 +24,8 @@ export interface DropdownItem {
 }
 
 export interface DropdownMenuProps {
+  /** Childrens to render as content in the dropdown */
+  children?: ReactNode;
```
Comment (pnagarjun16) [inline src/components/DropdownMenu/DropdownMenu.tsx:28]: Can we change it to Header?

**Review by pnagarjun16 — CHANGES_REQUESTED** [review]

File: src/components/DropdownMenu/DropdownMenu.tsx (hunk 4 above)
Comment (pnagarjun16) [reply src/components/DropdownMenu/DropdownMenu.tsx:28]: Change it to  `children?: ReactNode;`

**Review by pnagarjun16 — DISMISSED** [review]

**Review by pnagarjun16 — APPROVED** [review]

File: src/components/DropdownMenu/DropdownMenu.test.tsx [hunk 5]
```diff
@@ -133,15 +134,60 @@ describe('<DropdownMenu />', () => {
...
+      const { container } = render(
+        <Dropdown.DropdownMenu
+          {...DEFAULT_PROPS_MENU}
+          open={true}
+          defaultOpen={true}
+          header={<Dropdown.Header {...DEFAULT_PROPS_HEADER} className={mockClassName} item={mockText[0]} />}
+        />
+      );
+      expect(container.getElementsByClassName(mockClassName).length).toBe(1);
+      expect(screen.getByText(mockText[0])).toBeInTheDocument();
```
Comment (colbr) [inline src/components/DropdownMenu/DropdownMenu.test.tsx:158]: Can we separate the `className` test, and just do the `toBeInDocument` one here? Just so we're testing one thing at a time.

File: src/components/DropdownMenu/DropdownMenu.test.tsx [hunk 6]
```diff
@@ -133,15 +134,60 @@ describe('<DropdownMenu />', () => {
...
+    });
+  });
+});
+
+const renderHeader = (props?: Dropdown.HeaderProps) => render(<Dropdown.Header {...DEFAULT_PROPS_HEADER} {...props} />);
```
Comment (colbr) [inline src/components/DropdownMenu/DropdownMenu.test.tsx:163]: nit: brackets around function body

File: src/components/DropdownMenu/DropdownMenu.tsx [hunk 7]
```diff
@@ -87,3 +91,14 @@ export const DropdownMenu = forwardRef<HTMLDivElement, DropdownMenuProps>(
     );
   }
 );
+
+export interface HeaderProps {
+  /** Item to render inside the Header */
+  item: ReactNode;
```
Comment (colbr) [inline src/components/DropdownMenu/DropdownMenu.tsx:97]: Can we change `item` to `children`? This way we'll be able to use the component as `<DropdownHeader>Header Text</DropdownHeader>` - a little bit of a cleaner API. Tests will need to be updated for this

File: src/components/DropdownMenu/DropdownMenu.test.tsx [hunk 8]
```diff
@@ -133,15 +134,60 @@ describe('<DropdownMenu />', () => {
...
+    renderHeader({
+      item: (
+        <>
+          <MockComponent text={mockText[0]} />
+          <MockComponent text={mockText[1]} />
+        </>
+      )
+    });
+    mockText.forEach(item => expect(screen.getByText(item)).toBeInTheDocument());
+  });
```
Comment (colbr) [inline src/components/DropdownMenu/DropdownMenu.test.tsx:186]: We shouldn't need to test for `Fragment` - if the component test above passes the `Fragment` one will as well.

**Review by colbr — CHANGES_REQUESTED** [review]
Looks awesome @alejoviola 

Couple of requested changes, nothing major

**Review by colbr — APPROVED** [review]

**Pull request closed without merging**